### PR TITLE
[IR] Include metadata props in deserialization

### PR DESCRIPTION
**Pull request by justinchuby**
Deserialize metadata_props. Previously they were omitted.

### CODE REVIEW

File: onnxscript/ir/serde.py [hunk 1]
```diff
@@ -982,6 +997,8 @@ def serialize_tensor_into(
     if isinstance(from_, TensorProtoTensor):
         # Directly copy from the tensor proto if it is available
         tensor_proto.CopyFrom(from_.raw)
+        if from_.metadata_props:
+            _serialize_metadata_props_into(tensor_proto.metadata_props, from_.metadata_props)
         return
 
     tensor_proto.name = from_.name
```
Comment (justinchuby) [inline onnxscript/ir/serde.py:1004]: other tensors as well

File: onnxscript/ir/serde.py [hunk 2]
```diff
@@ -451,6 +458,7 @@ def deserialize_function(proto: onnx.FunctionProto) -> _core.Function:
         overload=getattr(proto, "overload", ""),
         graph=graph,
         attributes=typing.cast(List[_core.Attr], attributes),
+        metadata_props=deserialize_metadata_props(proto.metadata_props),
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/ir/serde.py:461]: SG. Should we just require 1.16?

[the remark this answers is not in the document]
Comment (BowenBao) [reply onnxscript/ir/serde.py:461]: idk, what's the convention for onnxscript? Do we provide BC support for previous ONNX versions?

[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/ir/serde.py:461]: Previous we do, but I also feel that we can follow torchvision and match onnxscript with an onnx version

[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/ir/serde.py:461]: @gramalingam

[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/ir/serde.py:461]: Bumped in #1391

**Review by titaiwangms — APPROVED** [review]

**Pull request merged**